app/scripts/metamask-controller.js

```diff
@@ -4454,6 +4456,38 @@ export default class MetamaskController extends EventEmitter {
 
       await this._addAccountsWithBalance(id);
 
+      const client = new KeyringInternalSnapClient({
+        messenger: this.controllerMessenger,
+        snapId: SOLANA_WALLET_SNAP_ID,
+      });
+      const keyring = await this.getSnapKeyring();
+      for (let index = 0; ; index++) {
+        const discovered = await client.discoverAccounts(
+          [SolScope.Mainnet, SolScope.Testnet, SolScope.Devnet],
+          id,
+          index,
+        );
+
+        if (discovered.length === 0) {
+          break;
+        }
+
+        await Promise.all(
```

Comment:
Do we want to use `allSettled` instead here? Since if one of the `createAccount` fails (for any reason), it will stop the entire discovery.

Logging a warning could be an option in that case, WDYT?